test/docs/promises.test.js

```diff
@@ -21,8 +21,6 @@ describe('promises docs', function () {
   });
 
   after(async function () {
-    mongoose.Promise = global.Promise;
```

Comment:
i personally think the default one should be cached outside of the individual tests and reset in a hook, because tests may fail in-between and so dont reset properly if the reset is after the failed point